src/shared/utilities/cliUtils.ts

```diff
@@ -314,6 +335,53 @@ async function installSsmCli(
         return finalPath
     }
 }
+export async function installSamCli(
+    tempDir: string,
+    progress: vscode.Progress<{ message?: string; increment?: number }>,
+    timeout: Timeout
+): Promise<string> {
+    progress.report({ message: msgDownloading })
+
+    const samInstaller = await downloadCliSource(awsClis['sam-cli'], tempDir, timeout)
+    const outDir = path.join(getToolkitLocalCliPath(), 'AWSSAMCLI')
+
+    getLogger('channel').info(`Installing AWS SAM CLI from ${samInstaller} to ${outDir}...`)
+    progress.report({ message: msgInstallingLocal })
+
+    return handleError(install())
+
+    async function install() {
+        if (process.platform !== 'linux') {
+            throw new InvalidPlatformError(`Platform ${process.platform} is not supported for CLI autoinstallation.`)
+        }
+        return installOnLinux()
+    }
+
+    async function installOnLinux() {
+        if (process.platform !== 'linux') {
+            throw new InvalidPlatformError(`Cannot use Linux installer on operating system: ${process.platform}`)
+        }
+        const dirname = path.dirname(samInstaller)
+        new admZip(samInstaller).extractAllTo(dirname, true)
+
+        const result = await new ChildProcess('sh', [
+            path.join(dirname, 'install'),
+            '-u',
+            '-i',
+            outDir,
+            '-b',
```

Comment:
short args like `-i` and `-b` are convenience aliaes for interactive usage. For non-interactive usage, it's usually helpful to use the long names (except perhaps for some very well-known commands).